R/ebp.R

```diff
@@ -313,38 +399,106 @@ ebp <- function(fixed,
   # MSE Estimation -------------------------------------------------------------
 
   if (MSE == TRUE) {
+    if(tf == FALSE){
+      # The function parametric_bootstrap can be found in script mse_estimation.R
+      mse_estimates <- parametric_bootstrap(
+        framework = framework,
+        point_estim = point_estim,
+        fixed = fixed,
+        transformation = transformation,
+        interval = interval,
+        L = L,
+        B = B,
+        boot_type = boot_type,
+        parallel_mode = parallel_mode,
+        cpus = cpus
+      )
 
-    # The function parametric_bootstrap can be found in script mse_estimation.R
-    mse_estimates <- parametric_bootstrap(
-      framework = framework,
-      point_estim = point_estim,
-      fixed = fixed,
-      transformation = transformation,
-      interval = interval,
-      L = L,
-      B = B,
-      boot_type = boot_type,
-      parallel_mode = parallel_mode,
-      cpus = cpus
-    )
 
 
+      ebp_out <- list(
+        ind = point_estim$ind,
+        MSE = mse_estimates,
+        transform_param = point_estim[c(
+          "optimal_lambda",
+          "shift_par"
+        )],
+        model = point_estim$model,
+        framework = framework[c(
+          "N_dom_unobs",
+          "N_dom_smp",
+          "N_smp",
+          "N_pop",
+          "smp_domains",
+          "smp_data",
+          "smp_domains_vec",
+          "pop_domains_vec"
+        )],
+        transformation = transformation,
+        method = "reml",
+        fixed = fixed,
+        call = call,
+        successful_bootstraps = NULL
+      )
+    }else{ #To me replaced when integrating the MSE for twofold
+      message("MSE is not yet defined for this method at this integration stage")
+      ebp_out <- list(
+        ind_Domain = point_estim$ind_Domain,
+        ind_Subdomain = point_estim$ind_Subdomain,
+        MSE = NULL,
+        transform_param = point_estim[c(
+          "optimal_lambda",
+          "shift_par"
+        )],
+        model = point_estim$model,
+        framework = framework[c(
+          "N_subdom_unobs",
+          "N_subdom_smp",
+          "N_dom_unobs",
+          "N_dom_smp",
+          "N_smp",
+          "N_pop",
+          "smp_subdomains",
+          "smp_domains",
+          "smp_data",
+          "smp_subdomains_vec",
+          "pop_subdomains_vec",
+          "smp_domains_vec",
+          "pop_domains_vec"
+        )],
+        transformation = transformation,
+        method = "reml",
+        fixed = fixed,
+        call = call,
+        successful_bootstraps = NULL
+      )
+    }
+
+
+  } else {
 
+  if( tf == TRUE){
     ebp_out <- list(
-      ind = point_estim$ind,
-      MSE = mse_estimates,
+      ind_Domain = point_estim$ind_Domain,
```

Comment:
It is not optimal to change the structure of the return of the function. So it is better that the return will still be ind and MSE. Our suggestion would be to have one data frame that is returned in ind and the domains and subdomains are named in the variable domain and an variable level is added which indicators if the row is domain or subdomain. So the domains and subdomains will be in one data frame. The other functions as estimators will serve the purpose to split the data frame for the analysis of interest.